base/abstractarray.jl

```diff
@@ -2159,12 +2159,17 @@ function hvcat_fill!(a::Array, xs::Tuple)
     if nr*nc != len
         throw(ArgumentError("argument count $(len) does not match specified shape $((nr,nc))"))
     end
-    k = 1
-    for i=1:nr
-        @inbounds for j=1:nc
-            a[i,j] = xs[k]
-            k += 1
+    i = 1
+    j = 1
+    ntuple(Val(len)) do k
```

Comment:
I think if we want to unroll this the standard way would be a generated function using `@nexprs`. We also might want to separate the homogeneous tuple case to still use a loop?